src/main/java/org/opensearch/knn/plugin/transport/TrainingModelRequest.java

```diff
@@ -103,6 +107,7 @@ public TrainingModelRequest(StreamInput in) throws IOException {
         this.maximumVectorCount = in.readInt();
         this.searchSize = in.readInt();
         this.trainingDataSizeInKB = in.readInt();
+        this.vectorDataType = VectorDataType.get(in.readOptionalString());
```

Comment:
This wont work in mixed cluster state - is that expected? If we want it to we cannot use readOptionalString we need to do version check.

Reply:
Can we use it like?
```
        if (IndexUtil.isVersionOnOrAfterMinRequiredVersion(in.getVersion(), KNNConstants.MODEL_VECTOR_DATA_TYPE_KEY)) {
            this.vectorDataType = VectorDataType.get(in.readOptionalString());
        } else {
            this.vectorDataType = VectorDataType.FLOAT;
        }
```